forbidigo/forbidigo.go

```diff
@@ -164,24 +170,40 @@ func (v *visitor) Visit(node ast.Node) ast.Visitor {
 		}
 		return v
 	case *ast.SelectorExpr:
+		selectorExpr = node
 	case *ast.Ident:
 	default:
 		return v
 	}
-	for _, p := range v.linter.patterns {
-		if p.pattern.MatchString(v.textFor(node)) && !v.permit(node) {
-			v.issues = append(v.issues, UsedIssue{
-				identifier: v.textFor(node),
-				pattern:    p.pattern.String(),
-				pos:        node.Pos(),
-				position:   v.fset.Position(node.Pos()),
-				customMsg:  p.msg,
-			})
+
+	literalText := v.textFor(node)
+	texts := []string{literalText}
+	if aliasedText := v.textWithPackage(selectorExpr); aliasedText != "" && aliasedText != literalText {
+		texts = append(texts, aliasedText)
+	}
+	if filename := v.filename(node); filename != "" {
+		numTexts := len(texts)
+		for i := 0; i < numTexts; i++ {
+			texts = append(texts, filename+":"+texts[i])
```

Comment:
```suggestion
		for _, text := range texts {
			texts = append(texts, filename+":"+text)
```

Reply:
I was unsure whether slices can be modified while iterating over them and still haven't found a definitive statement about that.

But as it seems to work and you suggested it, I'll trust that it is safe and change the code.

Reply:
I did miss that bit but per https://go.dev/ref/spec, "The range expression x is evaluated once before beginning the loop."